### PR TITLE
Search by criteria path 

### PR DESCRIPTION
**Pull request by AKatrin**
With this changes the search by path criteria using the Criteria class is implemented and refactored.
Where the path of directory is required.

### CODE REVIEW

File: src/main/java/com/jala/search/models/CriteriaSearch.java [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    /**
+     *
+     * @return the file name.
+     */
+    public String getFileName() {
+        return fileName;
+    }
+
+    /**
+     *
```
Comment (jpsandovaln) [inline src/main/java/com/jala/search/models/CriteriaSearch.java]: add documentation

File: src/main/java/com/jala/search/models/CriteriaSearch.java [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    /**
+     * This method allows to set the path sending a string.
+     * @param path of the directory.
+     */
+    public void setPath(String path) {
+        this.path = path;
+    }
+
+    /**
+     *
```
Comment (jpsandovaln) [inline src/main/java/com/jala/search/models/CriteriaSearch.java]: add documentation

File: src/main/java/com/jala/search/models/CriteriaSearch.java [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    /**
+     * Constructor that setting the path for then do a search.
+     * @param path
+     */
+    public CriteriaSearch(String path) {
+        this.path = path;
+    }
+
+    /**
+     *
```
Comment (jpsandovaln) [inline src/main/java/com/jala/search/models/CriteriaSearch.java]: add documentation

File: src/main/java/com/jala/search/models/CriteriaSearch.java [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+ * @version 0.0.1
+ * @author Areliez Vargas
+ */
+public class CriteriaSearch {
+
+    /**
+     * Criteria for path of the directory.
+     */
+    private String path;
+    /**
```
Comment (jpsandovaln) [inline src/main/java/com/jala/search/models/CriteriaSearch.java:27]: add one space between parameter for example:
private String path;

/**

File: src/main/java/com/jala/search/models/ISearchable.java [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+
+/**
+ * ISearchable interface to search by criteria.
+ * @version 0.0.1
+ * @author Areliez Vargas
+ */
+public interface ISearchable {
+
+    /**
+     *
```
Comment (jpsandovaln) [inline src/main/java/com/jala/search/models/ISearchable.java]: Add documentation

**Review by Regis18 — APPROVED** [review]
it is okey

**Comment by jpsandovaln** [comment]
when the last observations will be fixed this branch is ready to merge.

**Review by Regis18 — APPROVED** [review]

**Review by jpsandovaln — APPROVED** [review]

**Pull request merged**